src/WTG.BulkAnalysis.Core/Internal/SolutionProcessor.cs

```diff
@@ -165,6 +165,12 @@ async Task<ImmutableArray<Diagnostic>> GetProjectAnalyzerDiagnosticsAsync(Projec
 				.GetCompilationAsync(context.CancellationToken)
 				.ConfigureAwait(false);
 
+			if (compilation == null)
+			{
+				context.Log.WriteFormatted($"  - Skipping {project.Name} as it does not support compilation.", LogLevel.Warning);

```

Comment:
That's a new one to me. What projects don't support compilation? Shared projects?

Reply:
Shared projects won't even load (complains about not being able to identify the language from the `.shproj` file extension. This change is mostly because `.GetCompilationAsync' is now decorated as returning a nullable compilation and the [docs](https://docs.microsoft.com/en-us/dotnet/api/microsoft.codeanalysis.project.getcompilationasync?view=roslyn-dotnet-3.9.0#returns) say that it can return null when SupportsCompilation returns false. The [doc's for SupportsCompilation](https://docs.microsoft.com/en-us/dotnet/api/microsoft.codeanalysis.project.supportscompilation?view=roslyn-dotnet-3.9.0#Microsoft_CodeAnalysis_Project_SupportsCompilation) don't really provide any more detail on that.

Reply:
Interesting, something to keep an eye out for in future then.